src/event_form/steps/event-form-step5.controller.js

```diff
@@ -413,6 +431,109 @@ function EventFormStep5Controller($scope, EventFormData, eventCrud, udbOrganizer
       .then(markOrganizerAsCompleted, controller.showAsyncOrganizerError);
   };
 
+  function editingPrice(firstItem) {
+    if (firstItem === undefined) {
+      firstItem = false;
+    }
+
+    $scope.editPrice = true;
+    originalPrice = _.clone($scope.price);
+
+    if (firstItem && $scope.price.length === 0) {
+      $scope.price = [
+        {
+          category: 'base',
+          name: 'Basisprijs',
+          priceCurrency: 'EUR',
+          price: 0
+        }
+      ];
+    }
+
+    else if ($scope.price.length === 0) {
+      $scope.price = [
+        {
+          category: 'base',
+          name: 'Basisprijs',
+          priceCurrency: 'EUR',
+          price: ''
+        }
+      ];
+    }
+  }
+
+  function unsetPriceItemFree(key) {
+    $scope.price[key].price = '';
+  }
+
+  function setPriceItemFree(key) {
+    $scope.price[key].price = 0;
+  }
+
+  function deletePriceItem(key) {
+    $scope.price.splice(key, 1);
+  }
+
+  function showPriceDelete(key) {
+    return key !== 0;
+
+    // TODO when BE can accept empty price array
```

Comment:
Zou eventueel kunnen, maar dan moet ik logica in de template gaan stoppen. Op zich is dat nu geen probleem maar wanneer de backend een lege priceInfo accepteert is deze logica waarschijnlijk teveel om in de template te stoppen?